assets/src/components/imageUpload/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Image uploader component.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { data } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import './style.scss';
+
+/**
+ * Image select/upload button and modal.
+ */
+class ImageUpload extends Component {
+
+	/**
+	 * Constructor.
+	 */
+	constructor( props ) {
+		super( ...arguments );
+		this.state = {
+			frame    : false,
+		}
+	}
+
+	/**
+	 * Open the WP media modal.
+	 */
+	openModal = () => {
+		if ( this.state.frame ) {
+			this.state.frame.open();
+			return;
+		}
+
+		this.state.frame = wp.media( {
+			title: __( 'Select or upload image' ),
+			button: {
+				text: __( 'Select' )
+			},
+			library: {
+				type: 'image'
+			},
+			multiple: false
+		} );
+
+		this.state.frame.on( 'select', this.handleImageSelect );
+		this.state.frame.open();
+	}
+
+	/**
+	 * Update the state when an image is selected from the media modal.
+	 */
+	handleImageSelect = () => {
+		const { onChange } = this.props;
+		const attachment = this.state.frame.state().get( 'selection' ).first().toJSON();
```

Comment:
This is just a question, but do you think there's any wisdom in avoiding jQuery in our plugins?

Reply:
jQuery is definitely not going anywhere any time soon, but in general we don't need it. In this specific case, the media modal is built with jQuery, so there's no avoiding it.